tests/mocha/serializer_test.js

```diff
@@ -0,0 +1,1754 @@
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">1234567890a123a123a</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.Emoji = new SerializerTestCase(
+    'Emoji',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">😀👋🏿👋🏾👋🏽👋🏼👋🏻😀❤❤❤</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.Russian = new SerializerTestCase(
+    'Russian',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">ты любопытный кот</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.Japanese = new SerializerTestCase(
+    'Japanese',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">あなたは好奇心旺盛な猫です</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.Zalgo = new SerializerTestCase(
+    'Zalgo',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">z̴̪͈̲̜͕̽̈̀͒͂̓̋̉̍a̸̧̧̜̻̘̤̫̱̲͎̞̻͆̋ļ̸̛̖̜̳͚̖͔̟̈́͂̉̀͑̑͑̎ǵ̸̫̳̽̐̃̑̚̕o̶͇̫͔̮̼̭͕̹̘̬͋̀͆̂̇̋͊̒̽</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.ControlChars = new SerializerTestCase(
+    'ControlChars',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">&#x01;&#xa1;</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.testCases = [
+  Serializer.Mutations.Procedure.Names.Symbols,
+  Serializer.Mutations.Procedure.Names.EscapedSymbols,
+  Serializer.Mutations.Procedure.Names.SingleQuotes,
+  Serializer.Mutations.Procedure.Names.DoubleQuotes,
+  Serializer.Mutations.Procedure.Names.Numbers,
+  Serializer.Mutations.Procedure.Names.Emoji,
+  Serializer.Mutations.Procedure.Names.Russian,
+  Serializer.Mutations.Procedure.Names.Japanese,
+  Serializer.Mutations.Procedure.Names.Zalgo,
+  // TODO: Uncoment once #4945 is merged.
+  // Serializer.Mutations.Procedure.Names.ControlChars,
+];
+
+Serializer.Mutations.Procedure.testSuites = [
+  Serializer.Mutations.Procedure.Names,
+];
+
+Serializer.Mutations.testSuites = [
+  Serializer.Mutations.ControlsIf,
+  Serializer.Mutations.ListCreate,
+  Serializer.Mutations.Procedure,
+];
+
+Serializer.testSuites = [
+  Serializer.Attributes,
+  Serializer.Fields,
+  Serializer.Icons,
+  Serializer.Connections,
+  Serializer.Mutations,
+];
+
+var runSerializerTestSuite = (serializer, deserializer, testSuite) => {
```

Comment:
I know you're planning on refactoring this code out of core later, but for readability I would create a `serializer_test_helpers.js` which has the definitions for:
- `SerializerTestCase`
- `SerializerTestSuite`
- `runSerializerTestSuite`

and in this file:
- declare/create the test suites and test cases
- call `runSerializerTestSuite`

Reply:
I like this idea, but I think I'm going to leave it for now because everything is going to be moving into samples (including all of the test cases). I think if we wait we'll have a better chance of structuring it so that it makes the most sense for that package.